src/plant_species/species.repository.ts

```diff
@@ -0,0 +1,158 @@
+        data["OptimalCo2"] = updateSpeciesDTO.updateValues.optimalCo2;
+      }
+      if (updateSpeciesDTO.updateValues.optimalTemperature !== null) {
+        data["optimalTemperature"] =
+          updateSpeciesDTO.updateValues.optimalTemperature;
+      }
+      if (updateSpeciesDTO.updateValues.optimalHumidity !== null) {
+        data["optimalHumidity"] = updateSpeciesDTO.updateValues.optimalHumidity;
+      }
+      data["name"] = updateSpeciesDTO.updateValues.name;
+
+      return await this.prisma.plantSpecies.update({
+        where: { name: updateSpeciesDTO.nameToBeChanged },
+        data,
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        // The .code property can be accessed in a type-safe manner
+        if (e.code === "P2025") {
+          throw new HttpException(
+            "Species with this name does not exist",
+            HttpStatus.NOT_FOUND,
+          );
+        }
+        if (e.code === "P2002") {
+          throw new HttpException(
+            "Species with this name already exists",
+            HttpStatus.BAD_REQUEST,
+          );
+        }
+      }
+      throw e;
+    }
+  }
+
+  async getAllSpecies() {
+    const data = await this.prisma.plantSpecies.findMany({
+      include: {
+        PlantBatch: {
+          select: {
+            amount: true,
+          },
+        },
+      },
+    });
+    const transformedData = {};
+    data.forEach((item) => {
+      const {
+        name,
+        OptimalCo2,
+        optimalHumidity,
+        optimalTemperature,
+        PlantBatch,
+      } = item;
+      const totalPlants = PlantBatch.reduce(
+        (sum, batch) => sum + batch.amount,
+        0,
+      );
+
+      transformedData[name] = {
+        OptimalCo2,
+        optimalHumidity,
+        optimalTemperature,
+        totalPlants,
+      };
+    });
+    return transformedData;
+  }
+
+  async getSpeciesByName(name: string) {
+    const data = await this.prisma.plantSpecies.findUnique({
+      where: { name },
+      include: {
+        PlantBatch: {
+          select: {
+            id: true,
+            amount: true,
+            harvestDate: true,
+            plantingDate: true,
+          },
+        },
+      },
+    });
+
+    if (!data) {
+      throw new HttpException(
+        "Species with this name does not exist",
+        HttpStatus.NOT_FOUND,
+      );
+    }
+
+    const transformedData = {
+      name: data.name,
+      OptimalCo2: data.OptimalCo2,
+      optimalHumidity: data.optimalHumidity,
+      optimalTemperature: data.optimalTemperature,
+      totalPlants: data.PlantBatch.reduce(
+        (sum, batch) => sum + batch.amount,
+        0,
+      ),
+      PlantBatches: data.PlantBatch.reduce((result, batch) => {
+        result[`batchId-${batch.id}`] = {
+          amount: batch.amount,
+          harvestDate: batch.harvestDate,
+          plantingDate: batch.plantingDate,
+        };
+        return result;
+      }, {}),
```

Comment:
```suggestion
      }, {} as Record<string, { amount: number, harvestDate: Date, plantingDate: Date }>),
```

Add a type, avoid any.